solarforecastarbiter/io/reference_aggregates.py

```diff
@@ -0,0 +1,219 @@
+                'observation': 'Table Mountain Boulder CO dni',
+                'from': '2017-01-01T00:00Z',
+                'until': None
+            },
+            {
+                'site': 'NOAA SURFRAD Desert Rock NV',
+                'observation': 'Desert Rock NV dni',
+                'from': '2017-01-01T00:00Z',
+                'until': None
+            },
+            {
+                'site': 'NOAA SURFRAD Fort Peck MT',
+                'observation': 'Fort Peck MT dni',
+                'from': '2017-01-01T00:00Z',
+                'until': None
+            },
+            {
+                'site': 'NOAA SURFRAD Goodwin Creek MS',
+                'observation': 'Goodwin Creek MS dni',
+                'from': '2017-01-01T00:00Z',
+                'until': None
+            },
+            {
+                'site': 'NOAA SURFRAD Penn State Univ PA',
+                'observation': 'Penn State Univ PA dni',
+                'from': '2017-01-01T00:00Z',
+                'until': None
+            },
+            {
+                'site': 'NOAA SURFRAD Sioux Falls SD',
+                'observation': 'Sioux Falls SD dni',
+                'from': '2017-01-01T00:00Z',
+                'until': None
+            },
+        ]
+    }, {
+        'name': 'UO SRML Portland PV',
+        'description': 'Sum of a number of small PV systems in Portland, OR',
+        'variable': 'ac_power',
+        'aggregate_type': 'sum',
+        'interval_length': pd.Timedelta('1h'),
+        'interval_label': 'ending',
+        'timezone': 'Etc/GMT+8',
+        'observations': [
+            {
+                'site': 'UO SRML Portland OR PV 15 deg tilt',
+                'observation': 'Portland OR PV 15 deg tilt ac_power Unisolar',
+                'from': '2019-01-01T08:00Z',
+                'until': None,
+            },
+            {
+                'site': 'UO SRML Portland OR PV 30 deg tilt',
+                'observation': 'Portland OR PV 30 deg tilt ac_power Evergreen',
+                'from': '2019-01-01T08:00Z',
+                'until': None,
+            },
+            {
+                'site': 'UO SRML Portland OR PV 30 deg tilt',
+                'observation': 'Portland OR PV 30 deg tilt ac_power Photowatt',
+                'from': '2019-01-01T08:00Z',
+                'until': None,
+            },
+            {
+                'site': 'UO SRML Portland OR PV 30 deg tilt',
+                'observation': 'Portland OR PV 30 deg tilt ac_power Kaneka',
+                'from': '2019-01-01T08:00Z',
+                'until': None,
+            },
+            {
+                'site': 'UO SRML Portland OR PV 30 deg tilt',
+                'observation': 'Portland OR PV 30 deg tilt ac_power Sanyo',
+                'from': '2019-01-01T08:00Z',
+                'until': None,
+            },
+        ]
+    }
+]
+
+
+def generate_aggregate(observations, agg_def):
+    agg_obs = []
```

Comment:
```suggestion
    """Generate an aggregate object.
    
    Parameters
    ----------
    observations: list of datamodel.Observation
    agg_def: dict
        Text metadata to create a datamodel.Aggregate.
        'observation' field names will be matched against names
        of datamodel.Observation in ``observations``.
    
    Returns
    -------
    datamodel.Aggregate
    """
    agg_obs = []
```

Reply:
not sure if this should be a comment or a docstring, also not sure how to put this concisely, but something about replacing text description of observations in ``agg_def`` with observation objects obtained from the API